### ISSUE TITLE
Comparing signed to unsigned values in KokkosExp_MDRangePolicy.hpp

### ISSUE THREAD

**Issue by ibaned** [kokkos/kokkos#749]
Moving an issue from the Trilinos repo opened by @william76:

KokkosExp_MDRangePolicy.hpp was changed that added some new warnings to the build that was caught in the Werror build:

```
Trilinos/packages/kokkos/core/src/KokkosExp_MDRangePolicy.hpp:239:5: error: comparison between signed and unsigned integer expressions [-Werror=sign-compare]
Trilinos/packages/kokkos/core/src/KokkosExp_MDRangePolicy.hpp:245:7: error: comparison between signed and unsigned integer expressions [-Werror=sign-compare]
```

Offending code lines are: [239](https://github.com/trilinos/Trilinos/blob/develop/packages/kokkos/core/src/KokkosExp_MDRangePolicy.hpp#L239) and [245](https://github.com/trilinos/Trilinos/blob/develop/packages/kokkos/core/src/KokkosExp_MDRangePolicy.hpp#L245).

**Comment by ibaned** [comment]
I belive @ndellingwood  is the point of contact for MDRangePolicy

**Comment by ndellingwood** [comment]
Will make the quick change (add static_cast as appropriate) and PR to kokkos develop branch and to trilinos.

**Comment by ndellingwood** [comment]
Fix merged in with PR #753

**Comment by ndellingwood** [comment]
Identical fix also pushed to develop branch of Trilinos/kokkos
